proxy/src/telemetry/metrics/latency.rs

```diff
@@ -78,12 +87,50 @@ impl Histogram {
             .position(|max| &measurement <= max)
             .expect("latency value greater than u32::MAX; this shouldn't be \
                      possible.");
-        self.0[i] += 1;
+        self.buckets[i] += 1;
+
+        // It's time to play ~*Will It Overflow???*~
+        //
+        // If we make the fairly generous assumptions of 1-minute latencies
+        // and 1 million RPS per set of metric labels (i.e. per pod), that
+        // gives us:
+        //          600,000 (1 minute = 600,000 tenths-of-milliseconds)
+        //  x     1,000,000 (1 million RPS)
+        //  ---------------
+        //  600,000,000,000 (6e11) gain per second
+        //
+        // times the number of seconds in a day (86,400):
+        //      6e11 x 86400 = 5.184e16
+        //
+        // 18,446,744,073,709,551,615 is the maximum 64-bit unsigned integer.
+        //      1.8446744073709551615e19 / 5.184e16 = 355 (about 1 year)
+        //
+        // So at 1 million RPS with 1-minute latencies, the sum will overflow
+        // in about a year. We don't really expect a conduit-proxy process to
+        // run that long (or see this kind of load), but we can revisit this
+        // if supporting extremely long-running deployments becomes a priority.
+        //
+        // (N.B. that storing latencies in whole milliseconds rather than tenths
+        // of milliseconds would change the time to overflow to almost 10
+        // years.)
+        self.sum += measurement.0 as u64;
```

Comment:
Can we just wrap on overflow? most timeseries systems are designed to handle this.

i don't think it's safe to assume that a conduit proxy won't run for more than a year.

Reply:
Okay, would be happy to make this wrap instead --- @adleong said this was a reasonable assumption.